### PR TITLE
Homework + exercises Week 1

### PR DESCRIPTION
**Pull request by DenyGurzhiy**

### CODE REVIEW

File: week1/project/ecommerce/src/components/productList/categories.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+   const getCategoryName = (name) => name.slice(6 - name.length)
+
+   function setCategoryName(actCategory) {
+      props.useStateFunction((prevCategory) => {
+         previousCategoryRef.current = prevCategory;
+         activeCategoryRef.current = actCategory;
+         return actCategory
+      })
+   }
```
Comment (ObadaMoustafa) [inline week1/project/ecommerce/src/components/productList/categories.jsx]: why using all these constants ? you already got the setState from the props.

File: week1/project/ecommerce/src/components/productList/categories.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+   return (
+      <div className='categories-container'>
+         <div className='categories'>
+            {CategoriesList.map((category, index) => {
+               const categoryName = getCategoryName(category);
+               const className = setClassName(categoryName, previousCategoryRef.current, activeCategoryRef.current)
+               return <button className={className} key={index} id={index} onClick={() => { setCategoryName(categoryName) }}>{categoryName}</button>
+            })}
```
Comment (ObadaMoustafa) [inline week1/project/ecommerce/src/components/productList/categories.jsx]: it's better to separate the components .. I mean it's better to make component for Category then you can make map in Categories component.

**Review by ObadaMoustafa — COMMENTED** [review]
well done Denys .. I have some comments for you 
1- some Components need to be separated like categories and products
2- Using useRef without being nessecery 
3-  Components' names should start with capital letters

I want to say that finally, you solved the problem and the app is working well but these comments as refactoring.

so I can approve it for this week but take care you will face challenges in upcoming weeks so I do recommend you refactor from now

**Comment by DenyGurzhiy** [comment]
Hi,
Thank you very much for your review.
I've done all that you advised, but my all components name have first capital letter. Probably, you mean name of files with component should have first capital letter. Could you, please, clarify this issue? Thanks.

**Comment by ObadaMoustafa** [comment]
> Hi, Thank you very much for your review. I've done all that you advised, but my all components name have first capital letter. Probably, you mean name of files with component should have first capital letter. Could you, please, clarify this issue? Thanks.

React components must begin with a capital letter when rendered using JSX[[1]](https://iq.js.org/questions/react/why-should-component-names-start-with-capital-letter)[[2]](https://stackoverflow.com/questions/30373343/reactjs-component-names-must-begin-with-capital-letters)[[3]](https://www.designcise.com/web/tutorial/should-react-component-name-begin-with-capital-letter)[[4]](https://reactjs.org/docs/jsx-in-depth.html). This is because only HTML elements and SVG tags can begin with a lowercase letter[[1]](https://iq.js.org/questions/react/why-should-component-names-start-with-capital-letter). Capitalized component names are treated as React components, which get compiled into a direct reference to the named variable[[4]](https://reactjs.org/docs/jsx-in-depth.html). This convention helps React distinguish between HTML elements and React components[[5]](https://indepth.dev/posts/1499/why-component-identifiers-must-be-capitalized-in-react).

**Pull request closed without merging**